### PR TITLE
Add more in-depth documentation

### PR DESCRIPTION
**Pull request by AutonomicPerfectionist**
**Description**
This PR adds significantly more detailed documentation on how CMakeTest works internally.
It also updates the README and converts it to RST

**TODOs**

- [x] ~~Switch README to RST and use literal includes for the example~~ See below for context on feasibility
- [x] Add installation instructions

### CODE REVIEW

**Comment by AutonomicPerfectionist** [comment]
According to https://github.com/github/markup/issues/172 we can't use `literalinclude` in README.rst. There are workaround projects that involve scanning the file for `literalinclude` and then building a monolithic README with the contents inserted. This would require setting up a new workflow to regenerate this file whenever the source template or associated included files are changed

**Comment by ryanmrichard** [comment]
@AutonomicPerfectionist I think having a README like CMinx's is fine. We can just worry about code snippets in the documentation.

**Comment by AutonomicPerfectionist** [comment]
@ryanmrichard this one is ready for review

File: tests/tutorials/1_hello_world.cmake [hunk 1]
```diff
@@ -12,11 +12,12 @@ include(cmake_test/cmake_test)
 #access to all of the CMakeTest machinery. The testing philosophy of CMakeTest
 #closely follows that of the Catch2 C++ unit testing framework. In particular
 #this means each of your unit tests must be contained within an ``add_test``
-#block. The block starts with ``ct_add_test(<test_name>)`` and ends with
-#``ct_end_test()`` (all CMakeTest functions are namespaced with a prefix ``ct_``
+#block. The block starts with :obj:`~cmake_test/add_test.ct_add_test`
+#(all CMakeTest functions are namespaced with a prefix ``ct_``
 #to mitigate against collisions with similarly named functions from other
-#modules). ``ct_add_test`` and ``ct_end_test`` define a block scope. The CMake
-#code in this block scope will serve as the contents of the unit test.
```
Comment (ryanmrichard) [inline tests/tutorials/1_hello_world.cmake:19]: Originally I had a script which stripped these tutorial comments out and created an `.rst` file from it. This was before I knew of `.. literalinclude`. Point being, feel free to put these doc comments into a `.rst` file and then `.. literalinclude::` the test.

**Review by ryanmrichard — APPROVED** [review]

**Pull request merged**